library/ssl_tls13_server.c

```diff
@@ -207,16 +207,42 @@ static int ssl_tls13_offered_psks_check_identity_match_ticket(
      *
      * Servers MUST NOT use any value greater than 604800 seconds (7 days).
      *
+     * `ssl->conf->f_ticket_parse()` might return an illegal `ticket_lifetime`.
+     */
+    if (session->ticket_lifetime >
+        MBEDTLS_SSL_TLS1_3_MAX_ALLOWED_TICKET_LIFETIME) {
+        MBEDTLS_SSL_DEBUG_MSG(
+            3, ("Ticket lifetime exceeds maximum allowed value."));
+        goto exit;
+    }
+
+    /* RFC 8446 section 4.6.1
+     *
+     * The value of zero indicates that the ticket should be discarded
+     * immediately.
+     *
+     * `session->ticket_lifetime == 0` is a internal error for we won't send
+     * this kind of ticket.
+     *
+     */
+    if (session->ticket_lifetime == 0) {
+        MBEDTLS_SSL_DEBUG_MSG(
+            3, ("Ticket lifetime is zero"));
+        goto exit;
+    }
+
+    /*
```

Comment:
No need for that I think, we check that (lines 3062 to 3074) when sending tickets and the ticket lifetime is saved in ssl_ticket.c. Thus if ssl_ticket.c considers the ticket valid these constraints on the ticket lifetime are still fulfilled.